sound/soc/sof/intel/hda.c

```diff
@@ -1424,15 +1445,14 @@ static struct snd_soc_acpi_mach *hda_sdw_machine_select(struct snd_sof_dev *sdev
 			 * b) the NHLT table reports the presence of microphones
 			 */
 			if (!(mach->link_mask & GENMASK(3, 2))) {
-				const char *tplg_filename = mach->sof_tplg_filename;
 				int ret;
 
-				ret = dmic_topology_fixup(sdev, &tplg_filename, "", &dmic_num);
+				ret = dmic_topology_fixup(sdev, &tplg_filename, "", &dmic_num,
+							  tplg_fixup);
 				if (ret < 0)
 					return NULL;
-
-				pdata->tplg_filename = tplg_filename;
 			}
+			pdata->tplg_filename = tplg_filename;
```

Comment:
looks good to me @plbossart. I think if we explicitly set the pdata->tplg_filename on this line only when tplg_fixup is true, it will make it even better